tmt/plugins/__init__.py

```diff
@@ -417,6 +455,46 @@ def __len__(self) -> int:
     def __bool__(self) -> bool:
         return bool(self._plugins)
 
+    def create_decorator(
+        self, on_register: Optional[Callable[Concatenate[str, RegisterableT, P], None]] = None
+    ) -> Callable[Concatenate[str, P], Callable[[RegisterableT], RegisterableT]]:
```

Comment:
Ok I understand the issue that mypy is complaining. When you crated the `PluginRegistry` you essentially locked the type of `RegistrableT` that `create_decorator accepts, e.g. in `_PACKAGE_MANAGER_PLUGIN_REGISTRY` it is locked to `PackageManager[PackageManagerEngine]` not allowing derivatives of it like the `RpmOstree`. The solution is quite simple, use the `RegistrableT2` that is bound to `RegistrableT` in this spot (and probably the others)

Reply:
Can you provide a diff? `RegisterableT2` sounds like something I'd like to see first to understand where and how.

Reply:
Nope, still complains. I'm letting it be for now, it needs more work, and with `PackageManagerRegistry.provides("apk")` this may move elsewhere. I will also check whether we really need the `PackageManager` class to be generic, that would help as well, removing the hurdle.

Reply:
Yep it should complain since I didn't fix the it all around or added the appropriate type-hint to the instance. It is mostly for explaining the issue observed.

But there are more cleaner ways we can handle this still as mentioned in https://github.com/teemtee/tmt/pull/4109#pullrequestreview-3284069118